server/src/main/java/org/elasticsearch/action/admin/indices/create/TransportCreateIndexAction.java

```diff
@@ -82,6 +82,7 @@ protected void masterOperation(Task task, final CreateIndexRequest request, fina
                         throw systemIndices.netNewSystemIndexAccessException(threadPool.getThreadContext(), List.of(indexName));
                     }
                 } else {
+                    // NON_NET_NEW_ONLY should never be a possibility here, as it cannot be returned from getSystemIndexAccessLevel.
```

Comment:
There's an assert immediately below this line that the access level should be `NONE`, which includes that it shouldn't be `BACKWARDS_COMPATIBLE_ONLY` by implication. I also added asserts to other places this method is used, though, and a comment in `getSystemIndexAccessLevel` noting it's intentional that it can't be returned from that method: https://github.com/elastic/elasticsearch/pull/74803/commits/8bd9f816e332183b64a2fb2afe3dbccbe3cccaa1